lib/mogrify.ex

```diff
@@ -275,11 +276,22 @@ defmodule Mogrify do
     image_operator(image, "xc:#{color}")
   end
 
+  def add_option(image, option) do
+    validate_option(option)
+    custom(image, option.name, option.argument)
+  end
+
   def custom(image, action, options \\ nil) do
     %{image | operations: image.operations ++ [{action, options}]}
   end
 
   def image_operator(image, operator) do
     %{image | operations: image.operations ++ [{:image_operator, operator}]}
   end
+
+  defp validate_option(%Option{name: name, require_arg: true, argument: nil}) do
+    option_name = name |> String.replace("-", "")
+    raise ArgumentError, message: "the option #{option_name} need arguments. Be sure to pass arguments to option_#{option_name}(arg)"
```

Comment:
I'd rather extract this `raise` outside of this method, and just return a boolean value here. We could also rename it to `valid_option?` and just `raise` if it's not true. What do you think?

Reply:
:+1: Makes sense to have 2 separate methods in case `valid_option?/1` can be useful for another purpose.

Can rename `validate_option` to `validate_option!` to emphasize that it raises. And have it return the original option argument, for convenience.